src/sessions/softphone-session-handler.ts

```diff
@@ -107,11 +107,36 @@ export class SoftphoneSessionHandler extends BaseSessionHandler {
   }
 
   async handlePropose (pendingSession: IPendingSession): Promise<void> {
-    await super.handlePropose(pendingSession);
+    const isPrivAnswerAuto = pendingSession.privAnswerMode === 'Auto';
+    const eagerConnectionEstablishmentMode = this.sdk._config.eagerPersistentConnectionEstablishment;
+    const logInfo = { sessionId: pendingSession?.id, conversationId: pendingSession.conversationId };
+    
+    if (isPrivAnswerAuto) {
+      this.log('info', 'received a propose with privAnswerMode=true', logInfo);
+    }
+    
+    // if eagerPersistentConnectionEstablishment==='none' then we want to completely swallow the propose
+    const shouldIgnorePrivAnswerPropose = isPrivAnswerAuto && eagerConnectionEstablishmentMode === 'none';
+    if (shouldIgnorePrivAnswerPropose) {
+      this.log('info', 'eagerPersistentConnectionEstablishment is "none" so propose with privAnswerMode=true will be ignored', logInfo);
+      return;
+    }
+
+    const shouldAutoAnswerPrivately = isPrivAnswerAuto && eagerConnectionEstablishmentMode === 'auto';
+
+    // we want to emit the pendingSession event in all cases except when eagerConnectionEstablishmentMode === auto and this is a privAnswerMode call
+    if (!shouldAutoAnswerPrivately) {
+      await super.handlePropose(pendingSession);
+    } else if (shouldAutoAnswerPrivately) {
+      if (shouldAutoAnswerPrivately) {
```

Comment:
Maybe I'm misunderstanding but is this check redundant? `shouldAutoAnswerPrivately` needs to be true to get into this block, why would we need to check that again once we're inside?

Reply:
hmm yeah this is weird. I'll update it